test/jdk/javax/swing/JButton/bug4490179.java

```diff
@@ -31,61 +31,107 @@
                 button = new JButton("Button");
+                button.addActionListener(eventHandler);
+                button.addMouseListener(eventHandler);
+
+                frame = new JFrame("bug4490179");
                 frame.getContentPane().add(button);
-                button.addActionListener(e -> {
-                    if ((e.getModifiers() & InputEvent.BUTTON1_MASK)
-                            != InputEvent.BUTTON1_MASK) {
-                        System.out.println("Status: Failed");
-                        passed = false;
+
+                frame.addWindowFocusListener(new WindowAdapter() {
+                    @Override
+                    public void windowGainedFocus(WindowEvent e) {
+                        windowGainedFocus.countDown();
                     }
                 });
+
                 frame.pack();
                 frame.setLocationRelativeTo(null);
                 frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
                 frame.setVisible(true);
             });
+
+            if (!windowGainedFocus.await(1, SECONDS)) {
+                throw new RuntimeException("Window didn't gain focus");
+            }
             robot.waitForIdle();
-            robot.delay(1000);
+
             SwingUtilities.invokeAndWait(() -> {
-                pt = button.getLocationOnScreen();
-                buttonW = button.getSize().width;
-                buttonH = button.getSize().height;
+                Point location = button.getLocationOnScreen();
+                buttonCenter = new Point(location.x + button.getWidth() / 2,
+                                         location.y + button.getHeight() / 2);
             });
 
-            robot.mouseMove(pt.x + buttonW / 2, pt.y + buttonH / 2);
-            robot.waitForIdle();
+            robot.mouseMove(buttonCenter.x, buttonCenter.y);
+            System.out.println("Press / Release button 3");
             robot.mousePress(InputEvent.BUTTON3_DOWN_MASK);
             robot.mouseRelease(InputEvent.BUTTON3_DOWN_MASK);
 
+            System.out.println("Press button 1");
             robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            System.out.println("Press button 3");
             robot.mousePress(InputEvent.BUTTON3_DOWN_MASK);
+            System.out.println("Release button 3");
             robot.mouseRelease(InputEvent.BUTTON3_DOWN_MASK);
-            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
-            robot.delay(500);
 
-            if (!passed) {
-                throw new RuntimeException("Test Failed");
+            try {
+                if (!mouseButton3Released.await(1, SECONDS)) {
+                    throw new RuntimeException("Mouse button 3 isn't released");
+                }
+
+                robot.waitForIdle();
+
+                if (actionPerformed.await(100, MILLISECONDS)) {
+                    throw new RuntimeException("Action event triggered by releasing button 3");
```

Comment:
line here also should be put into 2 lines

Reply:
I'm reluctant about changing it… The PR is approved now. And the line still fits 100-column limit.

Yet here it wrapping the line has its value, the button number is at the end of the line.

Reply:
I talked to @azvegint and we decided it's good enough. Wrapping the lines would require re-review.